guides/messaging-high-availability.adoc

```diff
@@ -0,0 +1,227 @@
+in high availability topology with https://activemq.apache.org/components/artemis/documentation/latest/ha.html#shared-store[shared journal].
+
+The Apache ActiveMQ Artemis broker, embedded within WildFly, supports linking brokers together in primary-secondary pairs.
+This creates an active-passive HA design: under normal conditions, the primary broker handles all the workload, while the secondary node remains on standby,
+ready to take over if the primary broker fails. However, we will not start another WildFly server solely to support an inactive Artemis broker, instead we will configure an additional ActiveMQ Artemis broker
+on each WildFly instance to act as the secondary broker for the primary in the other WildFly. This setup results in two primary-secondary pairs, with each pair maintaining its own journal directory.
+
+image::artemis-collocated-topology.png[WildFly with ActiveMQ Artemis in collocated HA topology]
+
+If one WildFly server fails, the secondary ActiveMQ Artemis broker on the other WildFly server will activate and take over all duties.
+All HA enabled connections from the original primary broker will fail over to the secondary broker. The secondary broker will
+access all messages stored on the primary broker by loading them from the shared journal directory. Once the original WildFly server is restarted,
+the secondary broker will automatically switch back to standby mode, and all clients will fail back to the primary broker.
+
+[NOTE]
+--
+For simplicity, we will use just one machine to run WildFly instances and local directories for shared journals. In real world scenario those should be two separate machines with shared journal mount from shared file system (like NFS4).
+--
+
+[[prerequisites]]
+== Prerequisites
+
+To complete this guide, you need:
+
+* Roughly {prerequisites-time} minutes
+* JDK {jdk-minimal-version} installed with `JAVA_HOME` configured appropriately
+
+== Prepare shared journal directories
+
+Create two directories for each primary-secondary pair:
+
+[source, bash ,options="nowrap"]
+----
+mkdir messaging-journal-a
+mkdir messaging-journal-b
+----
+
+== Prepare WildFly Servers
+
+Now let's configure two WildFly servers with two primary-secondary pairs of embedded ActiveMQ Artemis brokers. ActiveMQ Artemis configured
+as primary on the first WildFly server and as secondary on the second WildFly server will use `messaging-journal-a`
+directory for their shared journal. Similarly primary broker on second WildFly server and secondary broker on
+the first WildFly server will use `messaging-journal-b` directory for their shared journal.
+
+[NOTE]
+--
+All ActiveMQ Artemis brokers must be in cluster so they know about each other and can provide cluster topology to connected clients. In case that any WildFly instance crashes, clients know the location of the secondary broker and are able to fail over.
```

Comment:
s/in cluster/in a cluster/g
s/In case that/In case/g

Reply:
When I read this I feel like there's info I'm missing; i.e. they need to be in a cluster but it doesn't say how that is achieved, so then I feel anxiety.  Perhaps add something like "The procedures in this guide ensure the brokers are properly clustered."